test_common/harness/testHarness.c

```diff
@@ -328,6 +320,24 @@ int runTestHarnessWithCheck( int argc, const char *argv[], int testNum, test_def
     }
 
     device = devices[choosen_device_index];
+	err = clGetDeviceInfo(device, CL_DEVICE_TYPE, sizeof(device_type), &device_type, NULL);
```

Comment:
You don't need to query the type of the device since the device was selected above from the type.

Reply:
The selection of the device type above this query was done by getting the environmental variables and looking at command line arguments, but in the case that none are set, I think it needs to be queried so that it doesn't simply run with CL_DEVICE_TYPE_DEFAULT if it is not specified.

Reply:
However, the query might need to happen before testharness checks for command line arguments and env vars

Reply:
Yes, there are indeed cases where the query is required but even with the query the code can't work in the general case as most tests are testing for equality between `gDeviceType` and a specific device type (like CPU or GPU) but the `CL_DEVICE_TYPE` query could return a combination of these (CPU | GPU | DEFAULT is apparently legal for example) so ultimately we will need to change all uses to check a single bit and fail or request command line info in non-sensical cases. I'll raise a spec bug to clarify whether that was intended.

Reply:
Let me know what specific action I should take to address that, since I think there should be something done about that

Reply:
Yes, it will take a while to fix all of that. @JohnKesapidesARM will add the query in #611.